### PR TITLE
CMake: Split omrvmstartup into separate library

### PR DESCRIPTION
**Pull request by dnakamura**
See #2530 for more discussion

Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

File: fvtest/vmtest/CMakeLists.txt [hunk 1]
```diff
@@ -30,16 +30,16 @@ target_link_libraries(omrvmtest
...
-	omrport
 	j9hashtable
 	j9avl
 	j9pool
-
+	omrvmstartup
+	${OMR_HOOK_LIB}
+	${OMR_GC_LIB}
+	${OMR_PORT_LIB}
+	${OMR_TRHREAD_LIB}
```
Comment (rwy0717) [inline fvtest/vmtest/CMakeLists.txt]: typo

Comment (dnakamura) [reply fvtest/vmtest/CMakeLists.txt]: well i guess they dont need the thread lib then 😄

File: omr/startup/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	trc_omrti
+)
+
+
+target_link_libraries(omrvmstartup
+	PRIVATE
+		omrcore
+		${OMR_HOOK_LIB}
+		${OMR_TRACE_LIB}
+)
```
Comment (rwy0717) [inline omr/startup/CMakeLists.txt]: Shouldn't these links be public?

Comment (dnakamura) [reply omr/startup/CMakeLists.txt]: i dont think so. plus on a static lib it doesnt make a difference anyway

Comment (rwy0717) [reply omr/startup/CMakeLists.txt]: I'm thinking more about include directory propagation.

Comment (dnakamura) [reply omr/startup/CMakeLists.txt]: I try to keep include directory propagation down to a minimum. I need to double check but I dont believe that you need any of  the hook or trace include directories in order to consume the apis exposed by this library

**Comment by charliegracie** [comment]
@genie-omr build all

File: fvtest/vmtest/CMakeLists.txt (hunk 1 above)
Comment (charliegracie) [reply fvtest/vmtest/CMakeLists.txt]: This may be the cause of the AIX compile failure :)

**Comment by charliegracie** [comment]
@dnakamura there are compile failures on AIX that will need to be cleaned up

**Comment by dnakamura** [comment]
@charliegracie  should be fixed

**Comment by youngar** [comment]
@genie-omr build all

**Pull request merged**